### PR TITLE
fix: add v4 shim due to experiment snapshot change

### PR DESCRIPTION
**Pull request by stoksc**
## Description
The format of the experiment snapshot was changed without a shim, causing paused experiments to fail on upgrade. This change adds a shim for v4 to v5 snapshots, and bumps the current version to v5.

## Test Plan
- [x] wrote a test
- [x] start an experiment on 0.17.7, pause it, upgrade. it should live. 

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/restore_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package internal
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestShimExperimentSnapshotV4(t *testing.T) {
```
Comment (rb-determined-ai) [inline master/internal/restore_test.go:9]: This looks good.

Now that we have a test with an embedded snapshot, can we write a test with a v5 snapshot, and ensure that it is loadable into the current system?  I feel like this sort of bug ought to be preventable.

File: master/internal/restore.go [hunk 2]
```diff
@@ -340,3 +341,62 @@ func shimExperimentSnapshotV2(snapshot []byte) ([]byte, error) {
 
 	return json.Marshal(experimentSnapshotV2)
 }
+
+// Version 3 => 4 had no shim, it was breaking.
+
+// Version 4 => 5 shims
```
Comment (rb-determined-ai) [inline master/internal/restore.go:347]: stray comment?  Or what is this saying?

**Review by rb-determined-ai — APPROVED** [review]

File: master/internal/restore_test.go (hunk 1 above)
Comment (stoksc) [reply master/internal/restore_test.go:9]: Yeah, good idea, let me take a shot at that.

File: master/internal/restore.go (hunk 2 above)
Comment (stoksc) [reply master/internal/restore.go:347]: eh, i'll remove these if they're confusing at all

**Pull request merged**